cfme/configure/configuration/__init__.py

```diff
@@ -294,230 +301,136 @@ def step(self):
-            uri: Hostname/IP address of the machine.
-            username: User name used for logging in (ftp, smb only).
-            password: Password used for logging in (ftp, smb only).
-
-        Usage:
-
-            log_credentials = ServerLogDepot.Credentials("nfs", "backup.acme.com")
-            log_credentials.update()
-            log_credentials = ServerLogDepot.Credentials(
-                "smb",
-                "foobar",
-                "backup.acme.com",
-                username="jdoe",
-                password="xyz",
-            )
-            log_credentials.update()
-
-        """
-        pretty_attrs = ['p_type', 'name', 'uri', 'username', 'password']
-
-        server_collect_logs = Form(
-            fields=[
-                ("type", {
-                    version.LOWEST: Select("select#log_protocol"),
-                    '5.5': AngularSelect('log_protocol')}),
-                ("name", {
-                    version.LOWEST: None,
-                    "5.4": Input("depot_name")}),
-                ("uri", Input("uri")),
-                ("user", Input("log_userid")),
-                ("password", MultiFill(Input("log_password"), Input("log_verify"))),
-            ]
-        )
-
-        validate = form_buttons.FormButton("Validate the credentials by logging into the Server")
-        # add FormButton because of inconsistency in attributes for Save buttons in cfme
-        save_button = {
-            version.LOWEST: form_buttons.save,
-            "5.4": form_buttons.angular_save,
-            "5.6": form_buttons.FormButton("Save Changes", ng_click="btnClick()")
-        }
-
-        def __init__(self, p_type, name, uri, username=None, password=None):
-            assert p_type in self.p_types.keys(), "{} is not allowed as the protocol type!".format(
-                p_type)
-            self.p_type = p_type
-            self.uri = uri
-            self.username = username
-            self.password = password
-            self.name = name
-
-        @cached_property
-        def p_types(self):
-            return version.pick({
-                version.LOWEST: dict(
-                    anon_ftp="Anonymous FTP",
-                    ftp="FTP",
-                    nfs="NFS",
-                    smb="Samba",
-                    dropbox="Red Hat Dropbox",
-                ),
-                "5.5": dict(
-                    anon_ftp=sel.ByValue("Anonymous FTP"),
-                    ftp=sel.ByValue("FTP"),
-                    nfs=sel.ByValue("NFS"),
-                    smb=sel.ByValue("Samba"),
-                    dropbox=sel.ByValue("Red Hat Dropbox")
-                )
-            })
+    def __init__(self, depot_type, depot_name=None, uri=None, username=None, password=None,
+                 appliance=None):
+        self.depot_name = depot_name
+        self.uri = uri
+        self.username = username
+        self.password = password
+        self.depot_type = depot_types[depot_type]
```

Comment:
**Optional**
Capture the potential KeyError here and re-raise with a message listing the available keys.